src/runmode-af-packet.c

```diff
@@ -355,6 +368,124 @@ static void *ParseAFPConfig(const char *iface)
+#else
+            SCLogError(SC_ERR_UNIMPLEMENTED, "Bypass set but eBPF support is not build-in");
+#endif
+        }
+    }
+
+    /* One shot loading of the eBPF file */
+    if (aconf->ebpf_filter_file) {
+#ifdef HAVE_PACKET_EBPF
+        int ret = EBPFLoadFile(aconf->iface, aconf->ebpf_filter_file, "filter",
+                               &aconf->ebpf_filter_fd, EBPF_SOCKET_FILTER);
+        if (ret != 0) {
+            SCLogError(SC_ERR_INVALID_VALUE,
+                       "Error when loading eBPF filter file");
+        }
+#else
+        SCLogError(SC_ERR_UNIMPLEMENTED, "eBPF support is not build-in");
+#endif
+    }
+
+    if (ConfGetChildValueWithDefault(if_root, if_default, "xdp-filter-file", &ebpf_file) != 1) {
+        aconf->xdp_filter_file = NULL;
+    } else {
+        SCLogInfo("af-packet will use '%s' as XDP filter file",
+                  ebpf_file);
+        aconf->xdp_filter_file = ebpf_file;
+        ConfGetChildValueBoolWithDefault(if_root, if_default, "bypass", &conf_val);
+        if (conf_val) {
+#ifdef HAVE_PACKET_XDP
+            SCLogConfig("Using bypass kernel functionality for AF_PACKET (iface %s)",
+                    aconf->iface);
+            aconf->flags |= AFP_XDPBYPASS;
+            RunModeAsksBypassManager();
+            BypassedFlowManagerRegisterCheckFunc(EBPFCheckBypassedFlowTimeout);
+#else
+            SCLogError(SC_ERR_UNIMPLEMENTED, "Bypass set but XDP support is not build-in");
+#endif
+        }
+#ifdef HAVE_PACKET_XDP
+        const char *xdp_mode;
+        if (ConfGetChildValueWithDefault(if_root, if_default, "xdp-mode", &xdp_mode) != 1) {
+            aconf->xdp_mode = XDP_FLAGS_SKB_MODE;
+        } else {
+            if (!strcmp(xdp_mode, "soft")) {
+                aconf->xdp_mode = XDP_FLAGS_SKB_MODE;
+            } else if (!strcmp(xdp_mode, "driver")) {
+                aconf->xdp_mode = XDP_FLAGS_DRV_MODE;
+            } else if (!strcmp(xdp_mode, "hw")) {
+                aconf->xdp_mode = XDP_FLAGS_HW_MODE;
+            } else {
+                SCLogError(SC_ERR_INVALID_VALUE,
+                           "Invalid xdp-mode value: '%s'", xdp_mode);
+            }
+        }
+#endif
+    }
+
+    /* One shot loading of the eBPF file */
+    if (aconf->xdp_filter_file) {
+#ifdef HAVE_PACKET_XDP
+        int ret = EBPFLoadFile(aconf->iface, aconf->xdp_filter_file, "xdp",
+                               &aconf->xdp_filter_fd, EBPF_XDP_CODE);
+        if (ret != 0) {
+            SCLogError(SC_ERR_INVALID_VALUE,
+                       "Error when loading XDP filter file");
+        } else {
+            ret = EBPFSetupXDP(aconf->iface, aconf->xdp_filter_fd, aconf->xdp_mode);
+            if (ret != 0) {
+                SCLogError(SC_ERR_INVALID_VALUE,
+                        "Error when setting up XDP");
+            }
+        }
+#else
+        SCLogError(SC_ERR_UNIMPLEMENTED, "XDP support is not build-in");
```

Comment:
should this be fatal?

Reply:
Currently, this patchset is performance only. So I would like an XDP config run on a non XDP aware system. But I'm not really opposed to get it fatal.